internal/bits/bit_array.go

```diff
@@ -409,6 +409,13 @@ func (bA *BitArray) UnmarshalJSON(bz []byte) error {
 	// Construct new BitArray and copy over.
 	numBits := len(bits)
 	bA2 := NewBitArray(numBits)
+	if bA2 == nil {
```

Comment:
Can't we simply check if `numBits` is 0?

Reply:
It defeats though the defensiveness because what happens in the future if NewBitArray changes? The point here is to defend against the final results of NewBitArray.